packages/product/integration-tests/__tests__/services/product-variant/index.ts

```diff
@@ -251,4 +251,82 @@ describe("ProductVariant Service", () => {
       )
     })
   })
+
+  describe("retrieve", () => {
+    beforeEach(async () => {
+      testManager = await TestDatabase.forkManager()
+
+      productOne = testManager.create(Product, {
+        id: "product-1",
+        title: "product 1",
+        status: ProductTypes.ProductStatus.PUBLISHED,
+      })
+
+      variantOne = testManager.create(ProductVariant, {
+        id: "test-1",
```

Comment:
suggestion(nit): Should we store the ids when we reuse them to prevent any type and simplify any changes in the future?